File: 1993-sum-of-all-subset-xor-totals/1993-sum-of-all-subset-xor-totals.cpp
class Solution {
public:
    int sum = 0, len = 0;
    int subsetXORSum(vector<int>& nums) {
        len = nums.size();
        for(int i = 0; i < len; i++) {
            cal(nums, i, 0);
        }

        return sum;
    }

    void cal(vector<int>& nums, int pos, int cur) {
        cur ^= nums[pos];
        sum += cur;
        for(int i = pos + 1; i < len; i++) {
            cal(nums, i, cur);
        }
    }
};

File: 1993-sum-of-all-subset-xor-totals/1993-sum-of-all-subset-xor-totals.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        sum = 0
        sizeOfNum = len(nums)

        def cal(nums, pos, cur) -> int:
            dsum = 0
            cur ^= nums[pos]
            dsum += cur
            for i in range(pos + 1, sizeOfNum) :
                dsum += cal(nums, i, cur)
            
            return dsum

        for i in range(sizeOfNum) :
            sum += cal(nums, i, 0)

        return sum